lang/semgrep-grammars/src/semgrep-dockerfile/test/corpus/semgrep.txt

```diff
@@ -65,3 +65,380 @@ RUN ["a", $X, "b"]
       (double_quoted_string)
       (semgrep_metavariable)
       (double_quoted_string))))
+
+=============================================
+Param metavariable
+=============================================
+
+FROM --platform=$PLATFORM debian
+ADD --chown=$USER_GROUP src dst
+
+---
+
+(source_file
+  (from_instruction
+    (param)
+    (image_spec
+      (image_name)))
+  (add_instruction
+    (param)
+    (path)
+    (path)))
+
+=============================================
+Image metavariable
+=============================================
+
+FROM $NAME:$TAG@$DIGEST
+FROM ${NAME}:${TAG}@${DIGEST}
+
+---
+
+(source_file
+  (from_instruction
+    (image_spec
+      (image_name
+        (expansion
+          (variable)))
+      (image_tag
+        (expansion
+          (variable)))
+      (image_digest
+        (expansion
+          (variable)))))
+  (from_instruction
```

Comment:
So looks like there is no difference between use $XXX or ${XXX}, both result in (expansion (variable)).
Is it intented?

Reply:
yes. I didn't write that, it's dockerfile syntax.